packages/components/notification/src/NotificationItem/NotificationItem.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useCallback } from 'react';
+import { cx } from 'emotion';
+import {
+  CheckCircleIcon,
+  ErrorCircleIcon,
+  WarningIcon,
+  CloseIcon,
+} from '@contentful/f36-icons';
+
+import { Button } from '@contentful/f36-button';
+import { TextLink } from '@contentful/f36-text-link';
+import { CommonProps, Flex, Box } from '@contentful/f36-core';
+import { Heading, Paragraph } from '@contentful/f36-typography';
+
+import { NotificationVariant, NotificationCta } from '../types';
+import { getStyles } from './NotificationItem.styles';
+
+export interface NotificationItemProps extends CommonProps {
+  /**
+   * Defines the styling of notification
+   * @default positive
+   */
+  variant?: NotificationVariant;
+  /**
+   * Defines if the close button should be rendered
+   * @default true
+   */
+  withCloseButton?: boolean;
+  /**
+   * Function that will be triggered when close button is clicked
+   */
+  onClose?: Function;
+  /**
+   * Title of the notification
+   */
+  title?: string;
+  /**
+   * Content of the notificaiton
+   */
+  children: React.ReactNode;
+  /**
+   * Label and text-link props of the CTA
+   */
+  cta?: Partial<NotificationCta>;
+}
+
+const _NotificationItem = (props: NotificationItemProps, ref) => {
+  const {
+    className,
+    children,
+    cta,
+    withCloseButton = true,
+    variant = 'positive',
+    onClose,
+    testId = 'cf-ui-notification',
+    title,
+    ...otherProps
+  } = props;
+
+  const styles = getStyles({ variant });
+
+  const renderIcon = useCallback(() => {
+    const size = title ? 'medium' : 'small';
+    const iconVariants = {
+      positive: <CheckCircleIcon variant={variant} size={size} />,
+      warning: <WarningIcon variant={variant} size={size} />,
+      negative: <ErrorCircleIcon variant={variant} size={size} />,
+    };
+
+    return <Box className={cx(styles.icon)}>{iconVariants[variant]}</Box>;
+  }, [variant, styles.icon, title]);
```

Comment:
Do we need to make a function and useCallback for this?

Would something like this work?

```jsx
const size = title ? 'medium' : 'small';
const iconVariants = {
  positive: <CheckCircleIcon variant={variant} size={size} />,
  warning: <WarningIcon variant={variant} size={size} />,
  negative: <ErrorCircleIcon variant={variant} size={size} />,
};

// ...

return (
    <Flex
      {...}
    >
      <Box className={cx(styles.icon)}>{iconVariants[variant]}</Box>
```

Reply:
TBH, I don't like this approach of using `useCallback` to create a node, they have zero impact on performance. And they only introduce confusion and bugs with dependency arrays. I saw this approach a couple of times already in our codebase.